iXBRLViewerPlugin/viewer/src/js/ixbrlviewer.js

```diff
@@ -156,8 +156,9 @@ export class iXBRLViewer {
         else {
             docTitle = "Inline Viewer";
         }
-        if ($('html').attr("lang") === undefined) {
-            $('html').attr("lang", "en-US");
+        if ($('html').attr("xml:lang") !== undefined ) {
+            $(iframe).contents().find('html').attr("xml:lang", $('html').attr("xml:lang"));
```

Comment:
The lang attribute was likely added to pass an automated accessibility checker that ensures a lang value is defined, but it didn’t verify whether the lang attribute accurately reflects the content.

There’s an open support ticket regarding this issue. I don’t have strong preferences for the final implementation, as long as we stop labeling French content as English.

Reply:
The `lang="en-US"` seems like the correct default for the application.  It was always correct prior to commit 9cb2512da0c435bf12198e21104ab015da03eba5, but perhaps it's not correct after `i18nInit` runs.  Perhaps `setLanguage` should also be updating `$('html').attr('lang')` to match the selected application language?

If we need to copy attributes into the iframe (I'm not sure if `xml:lang` has any meaning given that we know from the accessibility ticket that `xml:lang` wasn't good enough.. ?), it seems like we should copy all attributes like we do for `body` below.

Reply:
I tested all major desktop screen readers and observed three distinct behaviors:

1.	All languages are treated the same, using the same voice.
2.	Screen readers inspect the content to determine the language, ignoring both lang and xml:lang attributes (though these might be fallback options if the language can’t be inferred from the content).
3.	The lang attribute is used, while the xml:lang attribute is ignored.

In none of the tools did I find the xml:lang attribute being used. Additionally, the presence of content within an iframe or serving the document as XHTML made no difference to the behavior.

Based on these findings, I propose the following approach:

1.	For the non-stub viewer, if the xml:lang attribute is present, migrate it from the root html node to the html element of the report iframe.
2.	For both stub and non-stub viewers, set the lang attribute of the report iframe’s html element to the value of the report’s xml:lang attribute (if present).
3.	Set the lang attribute of the root html node to match the application UI language.
4.	If the report lacks an xml:lang value, set the lang attribute of the report iframe’s html element to an empty string. This effectively “resets” the language and prompts the screen reader to use its default voice.
5.	Within the inspector, override the lang attribute on taxonomy content elements with the selected taxonomy language.

Reply:
> 2. Screen readers inspect the content to determine the language, ignoring both lang
> 3. The lang attribute is used

These seem contradictory.  Can you clarify?

> 5. Within the inspector, override the lang attribute on taxonomy content elements with the selected taxonomy language.

I didn't follow this.  Could you give an example?

Otherwise, 1-4 sound good to me, thanks.

Reply:
> These seem contradictory.  Can you clarify?

They are. The listed behaviors are from different screen readers. A few examples:
1. NVDA - All languages are treated the same
2. macOS VoiceOver -  inspects the content to determine the language
3. JAWS - Uses the lang attribute

> I didn't follow this.  Could you give an example?

Taxonomy labels, references, etc. There are three languages in the viewer and the lang attributes should reflect that.

1. The report content (report iframe)
2. The selected application UI language (inspector panel)
3. The selected taxonomy language (labels )